gpt_index/indices/tree/base.py

```diff
@@ -71,8 +73,14 @@ def __init__(
             documents=documents,
             index_struct=index_struct,
             llm_predictor=llm_predictor,
+            text_splitter=text_splitter,
             **kwargs,
         )
+        # if not specified, use "smart" text splitter to ensure chunks fit in prompt
+        if text_splitter is None:
+            self._text_splitter = self._prompt_helper.get_text_splitter_given_prompt(
```

Comment:
_build_index_from_documents is called during the __init__ call which is in the line above, where _text_splitter isn't set yet. does this logic work?

Reply:
Oh yeah there's a much simpler way

Reply:
Hm actually this is turning out to be tricky -- will find a workaround